File: Csharp-NAutowired/LcdNumbers/DigitsSplitter.cs
using System.Collections.Generic;
using System.Linq;
using NAutowired.Core.Attributes;

namespace Org.Codecop.Lcdnumbers
{
    /// <summary>Splits into digits and converts digits to LCD digits.</summary>
    [Service]
    public class DigitsSplitter : IDigitsSplitter
    {
        [Autowired]
        private readonly INumeralSystem numeralSystem;
        [Autowired]
        private readonly IDigitFactory digitFactory;

        public virtual IList<Digit> Convert(int number)
        {
            IList<int> numeralDigits = numeralSystem.DigitsOf(number);
            return numeralDigits.Select(d => digitFactory.Create(d)).ToList();
        }
    }
}

File: Java-Spring/src/main/java/org/codecop/lcdnumbers/DigitsSplitter.java
package org.codecop.lcdnumbers;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.stereotype.Service;

/**
 * Splits into digits and converts digits to LCD digits.
 */
public class DigitsSplitter {

    private final NumeralSystem numeralSystem;
    private final DigitFactory digitFactory;

    public DigitsSplitter(NumeralSystem numeralSystem, DigitFactory digitFactory) {
        this.numeralSystem = numeralSystem;
        this.digitFactory = digitFactory;
    }

    public List<Digit> convert(int number) {
        List<Integer> numeralDigits = numeralSystem.digitsOf(number);

        return numeralDigits.stream(). //
                map(digitFactory::create). //
                collect(Collectors.toList());
    }
}